### ISSUE TITLE
Client-server design

### ISSUE THREAD

**Issue by 00sapo** [sbdchd/neoformat#351]
Hello,
I am trying to port JuliaFormatter in Neoformat. At now, it works, but Julia has slow startup-times. It can be improved by precompiling the package, but it would involve a bit of hacking by the user.

[JuliaFormatter.vim](https://github.com/kdheepak/JuliaFormatter.vim) uses a client-server architecture in which the server starts at the first call of the formatting command. The rest of the plugins is mainly concerned with IO operations that Neoformat provides out-of-the-box.

I am trying to build something similar by starting the server asynchrounously the first time that my function formatter is called (using `jobstart`) and the successive times I would like to return a command that send the buffer as stdin to my target job and copies its output. 

The result would be something like this:
```vim
let s:juliaserver_started = 0
function! neoformat#formatters#julia#juliaformatter() abort
    if s:juliaserver_started
        return {'exe': ..., 'args': ..., ...}
    else
        job_start('julia -e "# the server code"')
        let s:juliaserver_started = 1
    end
end
```

I have tried by using Unix commands (and `jobpid` to identify to server pid), but aside the fact it doesn't work on my system, it would also be not that portable.

Would it possible to return a call to a vim/lua function instead of an exe command?

**Comment by sbdchd** [comment]
Yeah could work but I think you’d have better luck / an easier time by writing something from scratch. Neoformat is pretty crufty as is
